### PR TITLE
Possible Rejoin Fix

### PR DESCRIPTION
**Pull request by ThatDan123**
-Might fix the odd occasion where you rejoin a game but end up on the new player join screen instead of rejoining body

-Caches lighting system state for admin tools

-Adds methods in player list to search for offline and online players as some methods should check both.

### CODE REVIEW

**Comment by DTraitor** [comment]
What happens if you join with the second client while the first client is on the server?

**Comment by ThatDan123** [comment]
The new joining player should get kicked for release builds

File: UnityProject/Assets/Scripts/Managers/PlayerList.cs [hunk 1]
```diff
@@ -302,7 +312,47 @@ public List<ConnectedPlayer> GetAllByUserID(string byUserID, bool includeOffline
...
+
+		return ConnectedPlayer.Invalid;
+	}
+
+	/// <summary>
+	/// Check logged off players
+	/// </summary>
+	/// <param name="condition"></param>
+	/// <returns></returns>
+	private ConnectedPlayer GetInternalLoggedOff(Func<ConnectedPlayer, bool> condition)
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Managers/PlayerList.cs:355]: This is identical to `GetInternalLoggedIn()`?

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Managers/PlayerList.cs:355]: dont worry im an idiot

**Pull request merged**